addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -388,13 +393,14 @@ def _rtc_leave_call(self, session_id=None):
             self.rtc_session_ids.unlink()
         else:
             self.channel_id._rtc_cancel_invitations(member_ids=self.ids)
-        if not self.channel_id.rtc_session_ids and self.channel_id.last_call_message_id:
-            # deducing information about the end of call through write_date
-            self.channel_id._message_update_content(
-                self.channel_id.last_call_message_id,
-                self.channel_id.last_call_message_id.body,
-                strict=False
+        if not self.channel_id.rtc_session_ids:
+            # sudo - dicuss.rtc.call.history: setting the end date of the call
+            # after it ends is allowed.
+            call_history_sudo = self.env["discuss.call.history"].sudo().search(
+                [("channel_id", "=", self.channel_id.id)], order="id DESC", limit=1
             )
+            call_history_sudo.end_dt = fields.Datetime.now()
```

Comment:
This might miss many flows, this should probably be in unlink of session instead.